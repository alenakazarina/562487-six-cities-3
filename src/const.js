export const BASE_URL = `https://htmlacademy-react-3.appspot.com/six-cities`;

export const SortType = {
  POPULAR: `Popular`,
  PRICE_TO_HIGH: `Price: low to high`,
  PRICE_TO_LOW: `Price: high to low`,
  TOP_RATED: `Top rated first`
};

export const OfferType = {
  house: `House`,
  room: `Private room`,
  apartment: `Apartment`,
  hotel: `Hotel`
};

export const AppRoute = {
  ROOT: `/`,
  OFFER: `/offer/:id`,
  FAVORITES: `/favorites`,
  LOGIN: `/login`
};

export const RATINGS = [
  {
    value: 5,
    title: `perfect`
  },
  {
    value: 4,
    title: `good`
  },
  {
    value: 3,
    title: `not bad`
  },
  {
    value: 2,
    title: `badly`
  },
  {
    value: 1,
    title: `terribly`
  }
];

export const MONTHS = [
  `January`, `February`, `March`, `April`,
  `May`, `June`, `July`, `August`,
  `September`, `October`, `November`, `December`
];

export const AuthStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const InitialState = {
  USER: {
    authStatus: `NO_AUTH`,
    user: {
      id: -1,
      email: ``,
      name: ``,
      isPro: false,
      avatarUrl: ``
    }
  },
  FAVORITES: {
    favorites: []
  },
  OFFER: {
    activeOffer: null,
    nearOffers: [],
    comments: []
  },
  OFFERS: {
    offers: [],
    activeCity: ``
  },
  ERRORS: {
    errorStatus: null
  }
};

