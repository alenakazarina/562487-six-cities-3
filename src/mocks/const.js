export const IMAGE_PATH = `./img/`;

const LOCATIONS_ZOOM = 13;

export const MONTHS = [
  `January`, `February`, `March`, `April`, `May`, `June`, `July`,
  `August`, `September`, `October`, `November`, `December`
];

const OfferTypes = {
  apartment: `Apartment`,
  room: `Private Room`,
  house: `House`,
  hotel: `Hotel`
};

export const OffersData = {
  maxRating: 5,
  types: Object.values(OfferTypes),
  titles: [
    `Canal View room`,
    `Nice, cozy, warm big bed apartment`,
    `Wood and stone place`,
    `Beautiful & luxurious apartment at great location`,
    `Cozy rooms in the city center`
  ],
  images: [
    `apartment-01.jpg`,
    `apartment-02.jpg`,
    `apartment-03.jpg`,
    `apartment-04.jpg`,
    `apartment-05.jpg`,
  ],
  amenities: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Cabel TV`, `Fridge`],
  description: `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
};

export const UserData = {
  names: [`Douglas`, `James`, `Matthew`, `John`, `Leo`, `Robert`, `Dirk`],
  avatarUrls: [
    `user-1.jpg`,
    `user-2.jpg`,
    `user-3.jpg`,
    `user-4.jpg`,
    `user-5.jpg`,
  ],
  comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`
};

export const LOCATIONS = [
  {
    name: `Paris`,
    location: {
      latitude: 48.856663,
      longitude: 2.351556,
      zoom: LOCATIONS_ZOOM
    }
  }, {
    name: `Cologne`,
    location: {
      latitude: 50.930779,
      longitude: 6.938399,
      zoom: LOCATIONS_ZOOM
    }
  }, {
    name: `Brussels`,
    location: {
      latitude: 50.851309,
      longitude: 4.351718,
      zoom: LOCATIONS_ZOOM
    }
  }, {
    name: `Amsterdam`,
    location: {
      latitude: 52.373057,
      longitude: 4.892557,
      zoom: LOCATIONS_ZOOM
    }
  }, {
    name: `Hamburg`,
    location: {
      latitude: 53.552645,
      longitude: 9.966287,
      zoom: LOCATIONS_ZOOM
    }
  }, {
    name: `Dusseldorf`,
    location: {
      latitude: 51.230569,
      longitude: 6.787428,
      zoom: LOCATIONS_ZOOM
    }
  }];
