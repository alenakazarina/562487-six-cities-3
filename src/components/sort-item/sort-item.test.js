import React from 'react';
import renderer from 'react-test-renderer';
import SortItem from './sort-item';
import {SortType} from '../../const';

const sortType = Object.values(SortType)[0];
const mockFn = () => {};

describe(`SortItem`, () => {
  it(`should render SortItem`, () => {
    const tree = renderer.create(
        <SortItem
          isActive={false}
          sortType={sortType}
          onSortTypeChange={mockFn}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`should render SortItem with active class`, () => {
    const tree = renderer.create(
        <SortItem
          isActive={true}
          sortType={sortType}
          onSortTypeChange={mockFn}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

