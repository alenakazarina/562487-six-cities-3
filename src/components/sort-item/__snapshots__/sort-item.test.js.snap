// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortItem should render SortItem 1`] = `
<li
  className="places__option "
  onClick={[Function]}
  tabIndex="0"
>
  Popular
</li>
`;

exports[`SortItem should render SortItem with active class 1`] = `
<li
  className="places__option places__option--active"
  onClick={[Function]}
  tabIndex="0"
>
  Popular
</li>
`;
